pkg/kubectl/cmd/edit.go

```diff
@@ -395,16 +383,30 @@ func getMapperAndResult(f cmdutil.Factory, args []string, options *resource.File
 	default:
 		return nil, nil, nil, "", fmt.Errorf("Not supported edit mode %q", editMode)
 	}
-	r := b.NamespaceParam(cmdNamespace).DefaultNamespace().
+
+	originalResult := b.NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, options).
 		ContinueOnError().
 		Flatten().
 		Do()
-	err = r.Err()
+	err = originalResult.Err()
 	if err != nil {
 		return nil, nil, nil, "", err
 	}
-	return mapper, resourceMapper, r, cmdNamespace, err
+
+	updatedResultGetter := func(data []byte) *resource.Result {
+		// resource builder to read objects from the original file or arg invocation
+		var b *resource.Builder
+		switch editMode {
+		case NormalEditMode:
```

Comment:
At some point we should pull all these conditionals into an interface with separate implementations and encapsulate all the special casing there.  Consider adding a TODO for some good samaritan to pick up.